.github/workflows/release.yml

```diff
@@ -0,0 +1,154 @@
+#################################################################################
+#  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+#  Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+#
+#  See the NOTICE file(s) distributed with this work for additional
+#  information regarding copyright ownership.
+#
+#  This program and the accompanying materials are made available under the
+#  terms of the Apache License, Version 2.0 which is available at
+#  https://www.apache.org/licenses/LICENSE-2.0.
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#
+#  SPDX-License-Identifier: Apache-2.0
+#################################################################################
+
+
+---
+name: "Release"
+
+on:
+  pull_request:
+    branches:
+      - main
+    types:
+      - closed
+
+  workflow_call:
+
+
+jobs:
+  # Gate
+  check-head:
+    name: "Check if repository is not fork AND head is release OR base is bugfix"
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'eclipse-tractusx/tractusx-edc' && (startsWith(github.ref_name, 'bugfix/') || startsWith(github.event.pull_request.head.ref, 'release/')) }}
```

Comment:
this looks complicated plus it will be evaluated for every PR closed. given that we're following a sort-of gitflow approach, why don't we release on the `release` branch and at the end we merge it back to `main` later? this will simplify things a little.

Reply:
Yes, it will be evaluated for every merged PR on `main` but it won't run because of the condition. And since this is a job-level conditional, github will just skip its execution and not throw any error (btw, that is the reason why I prefer job/action level conditionals vs everything script).

Like such.

![image](https://github.com/user-attachments/assets/f1f20744-2bda-4636-a4a9-5aa6e774a75c)


If we release from the `release` branch we can't create the tag from it, otherwise the release branch cannot be deleted.

I wouldn't say it looks complicated. Its actually very readable. It's just a veeeery long line 😅

Reply:
well, the tag can be created after the release branch has been merged into main. I think that's the typical gitflow approach

Reply:
I don't think I got your point then, sorry.
Your suggestion is to change the trigger of the release workflow to be the push on the `release` branch?

Reply:
Either having this workflow to run when a PR is closed on a `release` branch or making it manual as for the bugfix version. 
At the end of the release we could merge back to main if ref_name starts with `release` and snapshot version can be bumped.
The more the `release` and `bugfix` process are similar, the easier will be to maintain them

Reply:
The workflow is being triggered when a PR is closed on a `release` branch. If we really want to avoid the workflow to be evalutated on every closed PR to main, we can make it manual only.